frontend/test/metabase/scenarios/dashboard/actions-on-dashboards.cy.spec.js

```diff
@@ -99,6 +106,206 @@ const TEST_TABLE = "scoreboard_actions";
           expect(result.rows.length).to.equal(1);
         });
       });
+
+      it("adds an implicit create action to a dashboard and runs it", () => {
+        createModelFromTable(TEST_TABLE);
+        cy.get("@modelId").then(id => {
+          cy.request({
+            url: "/api/action",
+            method: "POST",
+            body: {
+              kind: "row/create",
+              name: "Create",
+              type: "implicit",
+              model_id: id,
+            },
+          });
+        });
+
+        cy.createDashboard({ name: `action packed dash` }).then(
+          ({ body: { id: dashboardId } }) => {
+            visitDashboard(dashboardId);
+          },
+        );
+
+        editDashboard();
+        cy.icon("click").click();
+        cy.get("aside").within(() => {
+          cy.findByPlaceholderText("Button text").clear().type("Create Team");
+          cy.button("Pick an action").click();
+        });
+
+        cy.findByRole("dialog").within(() => {
+          cy.findByText("Test Model").click();
+          cy.findByText("Create").click();
+          cy.button("Done").click();
+        });
+
+        saveDashboard();
+        cy.button("Create Team").click();
+
+        modal().within(() => {
+          cy.findByPlaceholderText("team_name").type("Zany Zebras");
+          cy.findByPlaceholderText("score").type("44");
+
+          cy.button("Save").click();
+        });
+
+        cy.wait("@executeAPI");
+
+        queryWritableDB(
+          `SELECT * FROM ${TEST_TABLE} WHERE team_name = 'Zany Zebras'`,
+          dialect,
+        ).then(result => {
+          expect(result.rows.length).to.equal(1);
+
+          expect(result.rows[0].score).to.equal(44);
+        });
+      });
+
+      it("adds an implicit update action to a dashboard and runs it", () => {
+        createModelFromTable(TEST_TABLE);
+        cy.get("@modelId").then(id => {
+          cy.request({
+            url: "/api/action",
+            method: "POST",
+            body: {
+              kind: "row/update",
+              name: "Update",
+              type: "implicit",
+              model_id: id,
+            },
+          });
+        });
+
+        cy.createDashboard({ name: `action packed dash` }).then(
+          ({ body: { id: dashboardId } }) => {
+            visitDashboard(dashboardId);
+          },
+        );
+
+        editDashboard();
+        setFilter("ID");
+        sidebar().within(() => {
+          cy.button("Done").click();
+        });
+
+        cy.icon("click").click();
+        cy.get("aside").within(() => {
+          cy.findByPlaceholderText("Button text").clear().type("Update Team");
+          cy.button("Pick an action").click();
+        });
+
+        cy.findByRole("dialog").within(() => {
+          cy.findByText("Test Model").click();
+          cy.findByText("Update").click();
+          cy.findAllByText(/ask the user/i)
+            .first()
+            .click();
+        });
```

Comment:
Seems like these three tests repeat the same steps (opening a dashboard, adding a new button, and assigning a new action to it). Would it make sense to extract this into a function? It'd be cool if I can update a single function and propagate these changes for all three kinds of actions here